lib/AST/DiagnosticConsumer.cpp

```diff
@@ -29,6 +29,17 @@ using namespace swift;
 
 DiagnosticConsumer::~DiagnosticConsumer() = default;
 
+DiagnosticInfo::FixIt::FixIt(CharSourceRange R, StringRef Str,
+                             ArrayRef<DiagnosticArgument> Args) : Range(R) {
+    // FIXME: Defer text formatting to later in the pipeline.
+    SmallString<16> Buffer;
+    llvm::raw_svector_ostream OS(Buffer);
+    DiagnosticEngine::formatDiagnosticText(OS, Str, Args,
+                                           DiagnosticFormatOptions::
+                                           formatForFixIts());
+    Text = OS.str();
```

Comment:
Since Text is already a std::string, we can avoid the extra copy by writing to that instead with a `raw_string_ostream`. (Or writing to a temporary `std::string` and then `std::move`-ing it.)